### PR TITLE
many: remove the concept of frameworks from the code

### PR DESCRIPTION
**Pull request by mvo5**
As discussed we will not support frameworks in 16.04. The new interface concept is what will be used instead. Remove most code that deals with frameworks (some systemd units are still left and need cleanup still).

### CODE REVIEW

**Comment by chipaca** [comment]
:+1:

File: integration-tests/tests/snap_find_test.go [hunk 1]
```diff
@@ -34,12 +34,12 @@ type searchSuite struct {
 }
 
 func (s *searchSuite) TestSearchFrameworkMustPrintMatch(c *check.C) {
```
Comment (stevenwilkin) [inline integration-tests/tests/snap_find_test.go]: This method name could do with being updated also

**Comment by stevenwilkin** [comment]
Commit 372660a needs to be removed

File: systemd/systemd_test.go [hunk 2]
```diff
@@ -232,8 +232,8 @@ WantedBy=multi-user.target
 `
 
 var (
-	expectedAppService  = fmt.Sprintf(expectedServiceFmt, "After=ubuntu-snappy.frameworks.target\nRequires=ubuntu-snappy.frameworks.target", ".mvo", "mvo", "Type=simple\n", arch.UbuntuArchitecture())
-	expectedFmkService  = fmt.Sprintf(expectedServiceFmt, "Before=ubuntu-snappy.frameworks.target\nAfter=ubuntu-snappy.frameworks-pre.target\nRequires=ubuntu-snappy.frameworks-pre.target", "", "", "Type=simple\n", arch.UbuntuArchitecture())
+	expectedAppService = fmt.Sprintf(expectedServiceFmt, "After=ubuntu-snappy.frameworks.target\nRequires=ubuntu-snappy.frameworks.target", ".mvo", "mvo", "Type=simple\n", arch.UbuntuArchitecture())
+
 	expectedDbusService = fmt.Sprintf(expectedServiceFmt, "After=ubuntu-snappy.frameworks.target\nRequires=ubuntu-snappy.frameworks.target", ".mvo", "mvo", "Type=dbus\nBusName=foo.bar.baz", arch.UbuntuArchitecture())
```
Comment (stevenwilkin) [inline systemd/systemd_test.go]: Empty line

**Comment by stevenwilkin** [comment]
:+1:

File: integration-tests/tests/snap_find_test.go (hunk 1 above)
Comment (mvo5) [inline integration-tests/tests/snap_find_test.go]: Thanks, good catch!

File: systemd/systemd_test.go (hunk 2 above)
Comment (mvo5) [inline systemd/systemd_test.go]: Fixed.

**Comment by mvo5** [comment]
@stevenwilkin Thanks for the review! 372660a is no longer part of the merge and I fixed the outlined issues.

**Comment by niemeyer** [comment]
LGTM!

**Comment by mvo5** [comment]
merge this please

**Comment by mvo5** [comment]
merge this please

**Comment by mvo5** [comment]
retest this please

**Comment by elopio** [comment]
You missed one integration tests: searchSuite.TestSearchFrameworkMustPrintMatch

Error: error: no snaps found for "hello-dbus-fwk"

**Pull request merged**